### PR TITLE
PR: Add explanation on the use of subsections in the Outline Explorer

### LINKED ISSUE #40: Document cell grouping
See spyder-ide/spyder#7107

### PR DESCRIPTION
**Pull request by ba-tno**
# Pull Request

## Pull Request Checklist

* [x] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md)
* [x] Based your PR on the latest version of the correct branch (master or 4.x)
* [x] Checked your writing carefully for correct English spelling, grammar, etc
* [x] Described your changes and the motivation for them below
* [ ] Noted what issue(s) this pull request resolves, creating one if needed

## Description of Changes
I added a paragraph on the use of subsections in the Editor to create a multi-level outline view in the Outline Explorer.

This is a retry of pull request #285, using the correct branch and upstream version, thanks for the guiding comments.

## Issue(s) Resolved

Fixes #40

### CODE REVIEW

File: doc/panes/editor.rst
Comment (CAM-Gerlach) [inline doc/panes/editor.rst:41]:
```suggestion
You can also create "subsections" by adding more ``%`` signs to the cell separator, e.g. ``# %%%`` to create a level 2 subsection, ``# %%%%`` for level 3, etc.
```

* Add clarifying "more"
* Tweak ending punctuation so sentence ends in a period.

File: doc/panes/editor.rst
Comment (CAM-Gerlach) [inline doc/panes/editor.rst:42]:
```suggestion
This displays multiple levels in the the :guilabel:`Outline Explorer`.
```

* Avoid word repetition ("create", "outline")
* Swap colon for period (per our convention for introducing figures)

File: doc/panes/editor.rst
Comment (CAM-Gerlach) [inline doc/panes/editor.rst:47]:
```suggestion
Note that this only affects how the outline is displayed; it has no impact on the functioning of the code cells.
```

* Use semicolon instead of comma to avoid a comma splice
* Swap "impact" and "effect" as it simplifies the sentence a bit and read a little cleaner, at least to my ear
* Avoid word repetition of "code"

File: doc/panes/editor.rst
Comment (CAM-Gerlach) [inline doc/panes/editor.rst:44]: Great job on the screenshot!

**Review by CAM-Gerlach — CHANGES_REQUESTED** [review]
Looking great @ba-tno ! Excellent job on the screenshot and the text is very good, I just had a few minor suggestions to address some minor copyediting nitpicks. Thanks!

Also, looks like you got everything sorted out with the branch issue; thanks (and sorry for all the extra work)! It looks like there are two stray commits in there somehow, but I can just fix that when merging. After everything's merged, to make sure your `master` and `4.x` branches are good to go, you can do `git fetch --all` , then run `git checkout <BRANCH> && git reset --hard upstream/<BRANCH>` for each branch.

**Comment by CAM-Gerlach** [comment]
Hey @ba-tno , thanks for the super quick reply! It looks like only the first suggestion got applied and others two were not; was that what you intended?

**Comment by ba-tno** [comment]
No, I thought I added three suggestions to the batch and committed that batch.... Let me try that again.

**Comment by ba-tno** [comment]
I've now applied the suggestion in separate commits, I hope this works now.

**Comment by CAM-Gerlach** [comment]
Thanks, sorry for that! Looks like everything is applied now, so its good to go from my end. Once the checks pass I'll approve, mark this as draft to remind us to fix the commits before merging, and tag another reviewer for a final look.

**Review by CAM-Gerlach — APPROVED** [review]
LGTM, thanks @ba-tno !

**Comment by CAM-Gerlach** [comment]
BTW, this fixes issue #40 ; didn't notice that before sorry.

**Review by juanis2112 — APPROVED** [review]
Thanks @ba-tno for your work. This is looking great for me. You can change the commits and merge now @CAM-Gerlach

**Comment by CAM-Gerlach** [comment]
Rebased to drop the wonky prior commits, optimize the image and fixup the review commits, and everything's merged now and the changes are live. Thanks again for your work and bearing with us, @ba-tno .

> It is also worth to have this in Spyder 5.x. What do you think is the best way of doing this @CAM-Gerlach?

@juanis2112 Same as in the main Spyder repo, after a PR is merged `4.x` gets merged to `master`, so always it picks up any changes in `4.x` (aside from merge conflicts and version-specific differences, which get manually resolved). This is why the message just above this comment appears (you might remember it from Spyder's repo too). In fact, it is this merge to master that triggers a rebuild of the site, as due to the architecture of `sphinx-multiversion`, all branches are built off the config in `master`.

Somewhat off topic, but it might be a good idea to have a maintainer guide where this information is documented. That would also be an appropriate place to document PR review level standards, build system information, the translation procedure, the release cycle/process, pre-commit and other such things that aren't really relevant to the average contributor but are important to know for maintainers. That way, if something happens to me again (or same to you), there would be a record of all this and clear step by step instructions that a new maintainer could follow (plus its helpful for us if we ourselves forget).

**Pull request merged**
